OBAWidget/Components /RefreshButton.swift

```diff
@@ -0,0 +1,49 @@
+//
+//  RefreshButton.swift
+//  OBAWidget
+//
+//  Created by Manu on 2024-10-18.
+//
+
+import SwiftUI
+import AppIntents
+import WidgetKit
+
+//  MARK: - RefreshWidgetIntent
+/// Since there's no built-in user interaction method for reloading widget timelines,
+/// this intent serves as a workaround to provide that functionality via a button.
+struct RefreshWidgetIntent: AppIntent {
+    static var title: LocalizedStringResource = "Refresh Widget"
+    
+    func perform() async throws -> some IntentResult {
+        // This forces the widget to refresh
+        WidgetCenter.shared.reloadAllTimelines()
```

Comment:
is Apple going to permit this? This feels like something they would expressly prohibit

Reply:
I think it should be fine for Apple. Since the refresh is triggered by the user to update the widget’s data only when needed, Apple mainly restricts frequent automatic refreshes. In our case, it is not like the user is going to refresh the widget every second when they want the bus information or every time they pickup their phone.